fs/config/ui.go

```diff
@@ -380,6 +381,7 @@ func backendConfig(ctx context.Context, name string, m configmap.Mapper, ri *fs.
 			if out.Option.Default == nil {
 				out.Option.Default = ""
 			}
+			s3.SetDefaultEndpointValueForIDriveE2(out, m)
```

Comment:
Can you move this call into `backend/s3/s3.go` too please?

Reply:
Please advise where is the right place to be moved to?

Reply:
Sorry, I misunderstood what your code was doing.

You need to call this in the `Config` callback which is called after all the initial config is done. The S3 backend hasn't got one, so you need to add one something like this:

```diff
 		Description: "Amazon S3 Compliant Storage Providers including AWS, Alibaba, Ceph, China Mobile, Cloudflare, ArvanCloud, Digital Ocean, Dreamhost, Huawei OBS, IBM COS, Lyve Cloud, Minio, Netease, RackCorp, Scaleway, SeaweedFS, StackPath, Storj, Tencent COS and Wasabi",
 		NewFs:       NewFs,
 		CommandHelp: commandHelp,
+		Config: func(ctx context.Context, name string, m configmap.Mapper, config fs.ConfigIn) (*fs.ConfigOut, error) {
+			switch config.State {
+			case "":
+				return nil, setDefaultEndpointValueForIDriveE2(ctx, m)
+			}
+			return nil, fmt.Errorf("unknown state %q", config.State)
+		},
 		Options: []fs.Option{{
 			Name: fs.ConfigProvider,
 			Help: "Choose your S3 provider.",
```

You can set values in the `m` at that point using `m.Set("endpoint", newValue)`.

You may want to stop the `endpoint` question being asked at all.

You can also ask the user questions too, but that requires adding more states.

Hopefully that is clear - let me know if it isn't and sorry for the confusion.

Reply:
Thank you very much for your help and patient!